src/operator/join.rs

```diff
@@ -39,6 +40,40 @@ where
     }
 }
 
+impl<P, I1> StreamIntegral<P, I1>
+where
+    P: Clone + 'static,
+{
+    pub fn join_incremental<K, V1, V2, V, W, F, I2, Z>(
```

Comment:
I find the placement of this method here a bit strange.
I think more natural would have to have a static method (no Self) in some Join operator that takes two StreamIntegral arguments. People will not find it here.